### ISSUE TITLE
core:math/noise noise.noise_2d(seed, {0,0}) is discontinuous and the same value regardless of seed.

### ISSUE THREAD

**Issue by thePHTest** [odin-lang/Odin#2862]
## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System & Odin Version:
 Odin: dev-2023-08-nightly:9453b238
        OS:   Windows 11 Professional (version: 22H2), build 22621.2283
        CPU:  AMD Ryzen Threadripper PRO 3955WX 16-Cores
        RAM:  65388 MiB

## Expected Behavior
New values for different seeds when using 0,0 for the coordinate. No discontinuity at 0.0 .

## Current Behavior

The result of `noise.noise_2d(seed, {0,0})` is always 0.00000000 or -0.00000000 regardless of the seed and leads to a discontinuity.

I haven't tested any of the other noise functions to see if they have the same issue

**Comment by Kelimion** [comment]
Good catch. Makes me wonder if that holds true for the [C noise library](https://github.com/KdotJPG/OpenSimplex2) it's based on.

**Comment by thePHTest** [comment]
Well there is this open issue for it so looks like it's in the original C noise library too:
https://github.com/KdotJPG/OpenSimplex2/issues/21

**Comment by Feoramund** [comment]
There was an update on the linked issue a few months after this was posted here. This is expected behavior for the origin to remain `0,0`.
